File: CH01_CodeSmells/CodeSmellTasks/LongParameterList.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace RefactoringToDesignPatterns.CH01_CodeSmells.CodeSmellTasks
{
    internal class LongParameterList
    {
        internal void RegisterProduct(Product product)
        {
            // Rejestracja produktu
            Console.WriteLine($"Product: {product.Name}, Category: {product.Category}, Price: {product.Price}, Stock: {product.Stock}, Supplier: {product.SupplierName}, Contact: {product.SupplierContact}");
        }
    }

    internal class Product
    {
        internal required string Name { get; set; }
        internal required string Category { get; set; }
        internal decimal Price { get; set; }
        internal int Stock { get; set; }
        internal required string SupplierName { get; set; }
        internal required string SupplierContact { get; set; }
    }
}

File: solutions_python/CH01_CodeSmells/LongParameterList.py
from dataclasses import dataclass

__all__ = ["LongParameterList"]

@dataclass
class Product:
    Name: str
    Category: str
    Price: float
    Stock: int
    SupplierName: str
    SupplierContact: str


class LongParameterList:
    @staticmethod
    def register_product(product: Product):
        print(
            f"Product: {product.Name}, Category: {product.Category}, Price: {product.Price}, 
            Stock: {product.Stock}, Supplier: {product.SupplierName}, Contact: {product.SupplierContact}"
        )